src/ripple/app/ledger/impl/LedgerMaster.cpp

```diff
@@ -945,8 +945,9 @@ LedgerMaster::checkAccept(uint256 const& hash, std::uint32_t seq)
         if (seq < mValidLedgerSeq)
             return;
 
-        valCount = app_.getValidations().numTrustedForLedger(hash);
-
+        auto validations = app_.getValidations().getTrustedForLedger(hash);
+        filterValsWithnUnl(validations, app_.validators().getnUnlNodeIDs());
```

Comment:
I don't really like that `filterValsWithnUnl` modifies the passed in argument, yet returns void. What if the functions `getTrustedForLedger()` and `currentTrusted()` accepted a bool parameter, such that if the parameter is true, the values are filtered before returning. I think that would be cleaner and would avoid having a function that modifies its parameter.

Reply:
I'd rather just have `negativeUnlFilter` be a function that takes a list and returns a filtered *copy*.

Reply:
I think returning a filtered copy is actually the cleanest way to do this, but I wasn't sure how expensive that copy was going to be. If it's not prohibitively expensive, I also prefer returning a filtered copy.

Reply:
Thanks, changed.